packages/backend/src/api/controllers/tvl/DetailedTvlController.ts

```diff
@@ -0,0 +1,94 @@
+import { Logger } from '@l2beat/shared'
+import { DetailedTvlApiResponse, UnixTime } from '@l2beat/shared-pure'
+
+import { ReportProject } from '../../../core/reports/ReportProject'
+import { AggregatedReportRepository } from '../../../peripherals/database/AggregatedReportRepository'
+import { AggregatedReportStatusRepository } from '../../../peripherals/database/AggregatedReportStatusRepository'
+import { ReportRepository } from '../../../peripherals/database/ReportRepository'
+import { ReportStatusRepository } from '../../../peripherals/database/ReportStatusRepository'
+import { getHourlyMinTimestamp } from '../utils/getHourlyMinTimestamp'
+import { getSixHourlyMinTimestamp } from '../utils/getSixHourlyMinTimestamp'
+import {
+  groupByProjectIdAndAsset,
+  groupByProjectIdAndTimestamp,
+} from './detailedTvl'
+import { generateDetailedTvlApiResponse } from './generateDetailedTvlApiResponse'
+
+export class DetailedTvlController {
+  constructor(
+    private readonly reportStatusRepository: ReportStatusRepository,
+    private readonly aggregatedReportRepository: AggregatedReportRepository,
+    private readonly aggregatedReportStatusRepository: AggregatedReportStatusRepository,
+    private readonly reportRepository: ReportRepository,
+    private readonly projects: ReportProject[],
+    private readonly logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+
+  async getDetailedTvlApiResponse(): Promise<
```

Comment:
I feel like this needs a test at least for the timestamps usage, ideally also for the response creation based on mock DB data